runtime/bcutil/ClassFileWriter.hpp

```diff
@@ -350,14 +352,33 @@ class ClassFileWriter {
 			_anonClassName = J9ROMCLASS_CLASSNAME(_romClass);
 			U_16 anonNameLength = J9UTF8_LENGTH(_anonClassName);
 			U_16 originalNameLength = anonNameLength - ROM_ADDRESS_LENGTH - 1;
+			U_8 *anonClassNameData = J9UTF8_DATA(_anonClassName);
+#if defined(J9VM_OPT_OPENJDK_METHODHANDLE)
+#define J9_INJECTED_INVOKER_CLASSNAME "InjectedInvoker"
+			/* ROM class format: <HOST_NAME>/InjectedInvoker/<ROM_ADDRESS_LENGTH>.
+			 * Search for InjectedInvoker in _anonClassName using the above format.
+			 * If found, reset the class name to "InjectedInvoker" in the class file.
+			 */
+			IDATA startIndex = anonNameLength - LITERAL_STRLEN(J9_INJECTED_INVOKER_CLASSNAME) - ROM_ADDRESS_LENGTH - 1;
+			U_8 *start = anonClassNameData + startIndex;
+			if ((startIndex >= 0)
+			&& (0 == memcmp(start, J9_INJECTED_INVOKER_CLASSNAME, LITERAL_STRLEN(J9_INJECTED_INVOKER_CLASSNAME)))
+			) {
+				_isInjectedInvoker = TRUE;
+				J9_DECLARE_CONSTANT_UTF8(anonClassName, J9_INJECTED_INVOKER_CLASSNAME);
```

Comment:
Declare this as: `J9_DECLARE_CONSTANT_UTF8(injectedInvokerClassname, "InjectedInvoker");` in a global scope.

Then you dont need `#define J9_INJECTED_INVOKER_CLASSNAME "InjectedInvoker"` at all

Reply:
Why global? Can't I put it right after the #if defined?

Reply:
If its in the method scope its allocated on stack and has to be re-initialized everytime. In this case that is not needed since the value never changes.

Reply:
But `J9_DECLARE_CONSTANT_UTF8` declares the var as static so it's allocated once in static memory.

Reply:
Ah yes, my mistake. Since its declared static it is init once. However the static init check is placed in the method as method statics are init lazily. Look at bindnatv.cpp for typical usage.

Reply:
I see `$$InjectedInvoker`... should I reuse the constant (get rid of the $$?) Also ROMClassBuilder.cpp already includes this file

Reply:
Ah missed the `$$` in that case you can leave it